### PR TITLE
Always return a valid pointer in Exception::what()

### PR DESCRIPTION
**Pull request by axic**
Fixes #2567 and #2832.

### CODE REVIEW

File: libdevcore/Exceptions.cpp [hunk 1]
```diff
@@ -27,7 +27,9 @@ char const* Exception::what() const noexcept
 	if (string const* cmt = comment())
 		return cmt->c_str();
```
Comment (chfast) [inline libdevcore/Exceptions.cpp:28]: `c_str()` is deprecated. Use `data()`.

File: libdevcore/Exceptions.cpp [hunk 2]
```diff
@@ -27,7 +27,9 @@ char const* Exception::what() const noexcept
...
 	else
```
Comment (chfast) [inline libdevcore/Exceptions.cpp:29]: You can skip `else`. Especially if the statement attached to `else` is so far away.

File: libdevcore/Exceptions.cpp (hunk 1 above)
Comment (axic) [reply libdevcore/Exceptions.cpp:28]: Can you address these in a new pull request?

File: libdevcore/Exceptions.cpp (hunk 2 above)
Comment (axic) [reply libdevcore/Exceptions.cpp:29]: Which one is far?

Comment (chfast) [reply libdevcore/Exceptions.cpp:29]: `return std::exception::what();` is 3 line below `else`.

**Review by chfast — APPROVED** [review]

**Comment by axic** [comment]
@chfast please create a new PR for your suggested changes

**Comment by chfast** [comment]
https://github.com/ethereum/solidity/pull/3009

**Pull request merged**